packages/type-utils/tests/TypeOrValueSpecifier.test.ts

```diff
@@ -395,15 +395,66 @@ describe('TypeOrValueSpecifier', () => {
       [
         `
           type SafePromise = Promise<number> & { __safeBrand: string };
-          type JoinedPromise = SafePromise & {};
+          type ResultType = { foo: 'bar' };
+          type Test = SafePromise & ResultType;
         `,
-        { from: 'file', name: ['JoinedPromise'] },
+        { from: 'file', name: ['ResultType'] },
```

Comment:
I think this test wasn't checking for intersections correctly, as it was checking the symbol of the "returned" type.